index.bs

```diff
@@ -346,20 +346,99 @@ interface mixin DatagramTransport {
+   of {{WebTransport/constructor(url, options)/url}}.
+1. If |parsedURL| is a failure, [=throw=] a {{SyntaxError}} exception.
+1. If |parsedURL| [=scheme=] is not `quic-transport` or `https`, [=throw=]
+   a {{SyntaxError}} exception.
+1. If |parsedURL| [=fragment=] is not null, [=throw=] a {{SyntaxError}} exception.
+1. Let |transport| be a newly constructed {{WebTransport}} object.
+1. Let |transport| have a
+   <dfn attribute for="WebTransport">\[[OutgoingStreams]]</dfn> internal slot
+   representing a sequence of {{OutgoingStream}} objects, initialized to empty.
+1. Let |transport| have a
+   <dfn attribute for="WebTransport">\[[ReceivedStreams]]</dfn> internal slot
+   representing a {{ReadableStream}} of {{IncomingStream}} objects, initialized
+   to empty.
+1. Let |transport| have a
+  <dfn attribute for="WebTransport">\[[ReceivedBidirectionalStreams]]</dfn>
+  internal slot representing a {{ReadableStream}} of {{IncomingStream}}
+  objects, initialized to empty.
+1. Let |transport| have a
+   <dfn attribute for="WebTransport">\[[WebTransportState]]</dfn> internal
+   slot, initialized to `"connecting"`.
+1. Let |transport| have a
+   <dfn attribute for="WebTransport">\[[SentDatagrams]]</dfn> internal slot
+   representing a {{WritableStream}} of {{Uint8Array}}s, initialized to empty.
+1. Let |transport| have a
+   <dfn attribute for="WebTransport">\[[ReceivedDatagrams]]</dfn> internal
+   slot representing a {{ReadableStream}} of {{Uint8Array}}s, initialized to
+   empty.
+1. If the scheme of |parsedURL| is `quic-transport`, [=in parallel=],
+   [=initialize WebTransport over QUIC=].
+1. If the scheme of |parsedURL| is `https`, [=in parallel=],
+   [=initialize WebTransport over HTTP=].
+1. Run these steps in parallel:
+1. Return |transport|.
+
+<div algorithm="initialize WebTransport over QUIC">
+To <dfn>initialize WebTransport over QUIC</dfn> for a given |transport| and
+|parsedURL|, do the following:
+
+1. Let |clientOrigin| be |transport|'s [=relevant settings object=]'s
+   [=origin=],
+   [serialized](https://html.spec.whatwg.org/multipage/origin.html#ascii-serialisation-of-an-origin).
+1. Establish a QUIC connection to the address identified by |parsedURL|
+   following the procedures in [[!WEB-TRANSPORT-QUIC]] section 3 and using
+   |clientOrigin| as the "origin of the client" referenced in section 3.2.1.
+   While establishing the connection, follow all of the parameters
+   specified in the {{WebTransport/constructor(url, options)/options}}.
+1. If the connection fails, set |transport|'s {{[[WebTransportState]]}}
+   internal slot to `"failed"` and abort these steps.
+1. Set |transport|'s {{[[WebTransportState]]}} internal slot to `"connected"`.
+
+</div>
+
+<div algorithm="initialize WebTransport over HTTP">
+To <dfn>initialize WebTransport over HTTP</dfn> for a given |transport| and
+|parsedURL|, do the following:
+
+1. Establish an HTTP/3 WebTransport session to the address identified by
```

Comment:
We should throw an exception if the `serverCertificateFingerprints` option was specified.